### PR TITLE
Make proxy endpoint configurable for s3 Http clients

### PR DESCRIPTION
**Pull request by flyrain**
cc @RussellSpitzer @szehon-ho @aokolnychyi @akshayakp97 @nastra

### CODE REVIEW

File: aws/src/main/java/org/apache/iceberg/aws/HttpClientProperties.java [hunk 1]
```diff
@@ -52,6 +52,13 @@ public class HttpClientProperties implements Serializable {
   public static final String CLIENT_TYPE_URLCONNECTION = "urlconnection";
 
   public static final String CLIENT_TYPE_DEFAULT = CLIENT_TYPE_APACHE;
+
+  /**
+   * Used to configure the proxy endpoint. Used by both {@link
+   * software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient.Builder} and {@link
+   * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}
+   */
+  public static final String PROXY_ENDPOINT = "http-client.proxy-endpoint";
```
[the remark this answers is not in the document]
Comment (amogh-jahagirdar) [reply aws/src/main/java/org/apache/iceberg/aws/HttpClientProperties.java:61]: Actually one thing that just occurred to me is that this proxy endpoint goes beyond S3, wouldn't it be hit for every service? Which is probably fine, and expected for a proxy endpoint but just wanted to make sure.

[the remark this answers is not in the document]
Comment (flyrain) [reply aws/src/main/java/org/apache/iceberg/aws/HttpClientProperties.java:61]: Will add it to the doc. I think it's fine to apply to other clients due to following two reasons.
1. If s3 needs it, most likely other clients need it as well.
2. They are not necessarily in the same catalog configuration.

**Comment by flyrain** [comment]
Thanks @amogh-jahagirdar for the review. Resolved your comments. Would you like to take another look?

**Review by amogh-jahagirdar — APPROVED** [review]

File: docs/docs/aws.md [hunk 2]
```diff
@@ -578,11 +578,12 @@ In contrast, URL Connection HTTP Client optimizes for minimum dependencies and s
 
 For more details of configuration, see sections [URL Connection HTTP Client Configurations](#url-connection-http-client-configurations) and [Apache HTTP Client Configurations](#apache-http-client-configurations).
 
-Configure the following property to set the type of HTTP client:
+Configurations for the HTTP client can be set in the catalog properties. The following are the available configurations:
```
Comment (nastra) [inline docs/docs/aws.md:581]:
```suggestion
Configurations for the HTTP client can be set via catalog properties. Below is an overview of available configurations:
```

**Review by nastra — APPROVED** [review]

**Comment by flyrain** [comment]
Thanks @amogh-jahagirdar and @nastra for the review.

**Pull request merged**